packages/xo-server-sdn-controller/src/index.js

```diff
@@ -717,6 +676,68 @@ class SDNController extends EventEmitter {
     }
   }
 
+  async _addHostToPoolNetworks(host) {
+    const xapi = host.$xapi
+
+    const tunnels = filter(xapi.objects.all, { $type: 'tunnel' })
+    for (const tunnel of tunnels) {
+      const accessPIF = await xapi._getOrWaitObject(tunnel.access_PIF)
+      if (accessPIF.host !== host.$ref) {
+        continue
+      }
+
+      const poolNetwork = find(this._poolNetworks, {
+        network: accessPIF.network,
+      })
+      if (poolNetwork == null) {
+        continue
+      }
+
+      if (accessPIF.currently_attached) {
+        continue
+      }
+
+      log.debug(
+        `Pluging PIF: '${accessPIF.device}' for host: '${
+          host.name_label
+        }' on network: '${accessPIF.$network.name_label}'`
+      )
+      try {
+        await xapi.call('PIF.plug', accessPIF.$ref)
+      } catch (error) {
+        log.error(
+          `XAPI error while pluging PIF: '${accessPIF.device}' on host: '${
+            host.name_label
+          }' for network: '${accessPIF.$network.name_label}'`
+        )
+      }
+
+      const starCenter = await xapi._getOrWaitObject(poolNetwork.starCenter)
+      await this._addHostToNetwork(host, accessPIF.$network, starCenter)
+    }
+  }
+
+  async _hostUnreachable(host) {
+    const poolNetworks = filter(this._poolNetworks, { starCenter: host.$ref })
+    for (const poolNetwork of poolNetworks) {
+      const network = await host.$xapi._getOrWaitObject(poolNetwork.network)
```

Comment:
Why not using `Xapi#getObjectByRef`?

`Xapi#_getOrWaitObject` is useful when the object might not be in the cache yet because just created.